File: 02 - Arrays/01 - Easy/02 - Find Second Largest/main.cpp
class Solution{
    public:
        int findSecondLargest(vector<int>& nums){
            int n = nums.size();

            int largest = INT_MIN;
            int secondLargest = INT_MIN;

            for(int i = 0; i < n; i++){
                if(nums[i] > largest){
                    secondLargest = largest;
                    largest = nums[i];
                }else if(nums[i] > secondLargest && nums[i] != largest){
                    secondLargest = nums[i];
                }
            }

            return secondLargest;
        }
};

File: 02 - Arrays/01 - Easy/02 - Find Second Largest/main.java
import java.util.*;

class Solution{
    public int findSecondLargest(List<Integer> nums){
        int largest = Integer.MIN_VALUE;
        int secondLargest = Integer.MIN_VALUE;

        int n = nums.size();

        for(int i = 0; i < n; i++){
            if(nums.get(i) > largest){
                secondLargest = largest;
                largest = nums.get(i);
            }else if(nums.get(i) > secondLargest && nums.get(i) != largest){
                secondLargest = nums.get(i);
            }
        }

        return secondLargest;
    }
}